dnsaas/api/v2/tests.py

```diff
@@ -1099,3 +1130,120 @@ def test_delete_nonexistent_record(self):
             'action': 'delete'
         }
         self._send_post_data_to_endpoint()
+
+
+class TestServiceField(BaseApiTestCase):
+    def setUp(self):
+        self.guest = get_user_model().objects.create_user(
+            'guest', 'guest@example.com', 'guest'
+        )
+        self.owner_with_access = get_user_model().objects.create_user(
+            'owner_with_access', 'owner_with_access@example.com',
+            'owner_with_access'
+        )
+        self.domain = DomainFactory(
+            name='example.com', owner=self.owner_with_access
+        )
+
+    def _record_for_update(self):
+        return RecordFactory(
+            domain=self.domain,
+            owner=self.owner_with_access,
+            type='A',
+            name='www.' + self.domain.name,
+            content='192.168.1.1',
+            service=None,
+        )
+
+    def test_create_record_saves_service(self):
+        self.client.login(
+            username='owner_with_access', password='owner_with_access'
+        )
+        service = ServiceFactory()
+
+        response = self.send_post(
+            reverse('api:v2:record-list'),
+            {
+                'type': 'A',
+                'domain': self.domain.id,
+                'name': 'example.com',
+                'content': '192.168.0.1',
+                'service': service.id,
+
+            },
+        )
+
+        record = Record.objects.get(pk=response.data['id'])
+        self.assertEqual(record.service.id, service.id)
+        self.assertEqual(response.data['service'], service.id)
+
+    def test_create_record_raise_error_when_no_service(self):
+        self.client.login(
+            username='owner_with_access', password='owner_with_access'
+        )
+
+        response = self.send_post(
+            reverse('api:v2:record-list'),
+            {
+                'type': 'A',
+                'domain': self.domain.id,
+                'name': 'example.com',
+                'content': '192.168.0.1',
+            },
+        )
+
+        self.assertEqual(
+            response.data['service'], ['This field is required.']
+        )
+
+    def test_edit_record_raise_error_when_service_is_empty(self):
+        self.client.login(
+            username='owner_with_access', password='owner_with_access'
+        )
+        record = self._record_for_update()
+
+        response = self.send_patch(
```

Comment:
imo it's bad idea to require some field in patch method (that's the idea of patch - partial update - i'd require it only in put request)